### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by AinaArd**

### CODE REVIEW

**Review by amourety — APPROVED** [review]

**Review by Dimanaux — CHANGES_REQUESTED** [review]
Reject

**Comment by amourety** [comment]
Nice

File: src/main/java/com/github/javalab/javaiaas/security/details/UserDetailsImpl.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+@Data
+@Component
+public class UserDetailsImpl implements UserDetails, UserDetailsService {
+
+    private User user;
+    private String token;
+    private UsersRepository usersRepository;
+
+    UserDetailsImpl(User user) {
```
Comment (amourety) [inline src/main/java/com/github/javalab/javaiaas/security/details/UserDetailsImpl.java]: sada

Comment (Dimanaux) [inline src/main/java/com/github/javalab/javaiaas/security/details/UserDetailsImpl.java:26]: Почему???

Comment (Dimanaux) [inline src/main/java/com/github/javalab/javaiaas/security/details/UserDetailsImpl.java]: Why package protected?

File: src/main/java/com/github/javalab/javaiaas/security/filters/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    private JwtAuthentication authentication;
+
+    @Autowired
+    public JwtAuthenticationFilter(JwtAuthenticationProvider provider, JwtAuthentication authentication) {
+        this.provider = provider;
+        this.authentication = authentication;
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
```
Comment (Dimanaux) [inline src/main/java/com/github/javalab/javaiaas/security/filters/JwtAuthenticationFilter.java]: Do you need this method?

File: src/main/java/com/github/javalab/javaiaas/security/filters/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        } else {
+            authentication.setAuthenticated(false);
+        }
+        filterChain.doFilter(servletRequest, servletResponse);
+    }
+
+    @Override
+    public void destroy() {
+    }
+}
```
Comment (Dimanaux) [inline src/main/java/com/github/javalab/javaiaas/security/filters/JwtAuthenticationFilter.java]: Newline please

File: src/main/java/com/github/javalab/javaiaas/security/filters/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            SecurityContextHolder.getContext().setAuthentication(provider.authenticate(authentication));
+        } else {
+            authentication.setAuthenticated(false);
+        }
+        filterChain.doFilter(servletRequest, servletResponse);
+    }
+
+    @Override
+    public void destroy() {
+    }
```
Comment (Dimanaux) [inline src/main/java/com/github/javalab/javaiaas/security/filters/JwtAuthenticationFilter.java]: Do you need this method?

File: src/main/java/com/github/javalab/javaiaas/security/filters/JwtRequestFilter.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        chain.doFilter(request, response);
+    }
+
+    private void setAuthAndDetails(String username, String jwtToken, HttpServletRequest request) {
+        if (username != null && SecurityContextHolder.getContext().getAuthentication() == null) {
+            UserDetails userDetails = this.userDetails.loadUserByUsername(username);
+            if (jwtTokenUtil.validateToken(jwtToken, userDetails)) {
+                UsernamePasswordAuthenticationToken authToke = new UsernamePasswordAuthenticationToken(
+                        userDetails, null, userDetails.getAuthorities());
+                authToke
```
Comment (Dimanaux) [inline src/main/java/com/github/javalab/javaiaas/security/filters/JwtRequestFilter.java]: toke?

File: src/main/java/com/github/javalab/javaiaas/security/providers/JwtAuthenticationProvider.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            throw new BadCredentialsException("Incorrect Token");
+        }
+        return jwtAuthentication;
+    }
+
+    @Override
+    public boolean supports(Class<?> authentication) {
+        return JwtAuthentication.class.equals(authentication);
+    }
+}
```
Comment (Dimanaux) [inline src/main/java/com/github/javalab/javaiaas/security/providers/JwtAuthenticationProvider.java]: newline

**Review by amourety — APPROVED** [review]

File: src/main/java/com/github/javalab/javaiaas/services/UsersServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    private String createToken(User user) {
+        return Jwts.builder()
+                .claim("login", user.getLogin())
+                .claim("id", user.getId())
+                .signWith(SignatureAlgorithm.HS512, secretKey)
+                .compact();
+    }
+
```
Comment (Dimanaux) [inline src/main/java/com/github/javalab/javaiaas/services/UsersServiceImpl.java]: Move this line to the end of file src/main/java/com/github/javalab/javaiaas/security/util/JwtTokenUtil.java

**Review by Dimanaux — APPROVED** [review]

File: src/main/java/com/github/javalab/javaiaas/services/UsersServiceImpl.java (hunk 7 above)
Comment (AinaArd) [reply src/main/java/com/github/javalab/javaiaas/services/UsersServiceImpl.java]: fixed

File: src/main/java/com/github/javalab/javaiaas/security/providers/JwtAuthenticationProvider.java (hunk 6 above)
Comment (AinaArd) [reply src/main/java/com/github/javalab/javaiaas/security/providers/JwtAuthenticationProvider.java]: fixed

File: src/main/java/com/github/javalab/javaiaas/security/filters/JwtRequestFilter.java (hunk 5 above)
Comment (AinaArd) [reply src/main/java/com/github/javalab/javaiaas/security/filters/JwtRequestFilter.java]: fixed

File: src/main/java/com/github/javalab/javaiaas/security/filters/JwtAuthenticationFilter.java (hunk 4 above)
Comment (AinaArd) [reply src/main/java/com/github/javalab/javaiaas/security/filters/JwtAuthenticationFilter.java]: deleted

File: src/main/java/com/github/javalab/javaiaas/security/filters/JwtAuthenticationFilter.java (hunk 3 above)
Comment (AinaArd) [reply src/main/java/com/github/javalab/javaiaas/security/filters/JwtAuthenticationFilter.java]: fixed

File: src/main/java/com/github/javalab/javaiaas/security/filters/JwtAuthenticationFilter.java (hunk 2 above)
Comment (AinaArd) [reply src/main/java/com/github/javalab/javaiaas/security/filters/JwtAuthenticationFilter.java]: deleted

File: src/main/java/com/github/javalab/javaiaas/security/details/UserDetailsImpl.java (hunk 1 above)
Comment (AinaArd) [reply src/main/java/com/github/javalab/javaiaas/security/details/UserDetailsImpl.java]: fixed

File: src/main/java/com/github/javalab/javaiaas/security/details/UserDetailsImpl.java [hunk 8]
```diff
@@ -23,7 +23,7 @@
     private String token;
     private UsersRepository usersRepository;
 
-    UserDetailsImpl(User user) {
+    private UserDetailsImpl(User user) {
```
Comment (Dimanaux) [inline src/main/java/com/github/javalab/javaiaas/security/details/UserDetailsImpl.java]: Why private?

**Review by Dimanaux — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/javalab/javaiaas/security/details/UserDetailsImpl.java (hunk 8 above)
Comment (AinaArd) [reply src/main/java/com/github/javalab/javaiaas/security/details/UserDetailsImpl.java]: changed

**Review by Dimanaux — APPROVED** [review]

**Pull request merged**